application/src/main/java/org/thingsboard/server/service/edge/rpc/constructor/OtaPackageMsgConstructor.java

```diff
@@ -62,7 +64,11 @@ public OtaPackageUpdateMsg constructOtaPackageUpdatedMsg(UpdateMsgType msgType,
             builder.setDataSize(otaPackage.getDataSize());
         }
         if (otaPackage.getData() != null) {
-            builder.setData(ByteString.copyFrom(otaPackage.getData().array()));
+            try {
+                builder.setData(ByteString.copyFrom(otaPackage.getData().readAllBytes()));
```

Comment:
This is an interesting place. Why do we need data in the OtaPackageUpdateMsg? It seems to load the entire file.

Reply:
it does, should I exclude it?